### ISSUE TITLE
Can not edit/upgrade the app if the original chart is hidden by the rancher-version annotation 

### ISSUE THREAD

**Issue by cbron** [rancher/rancher#34919]
Backend issue for: https://github.com/rancher/dashboard/issues/4276

Look into filtering the chart _only_ from the index.

**Comment by cbron** [comment]
Once the UI changes merge this will be ready to test

**Comment by cbron** [comment]
This can be tested along with https://github.com/rancher/rancher/issues/34875

**Comment by brendarearden** [comment]
### Root cause 
All api calls to the index, info, chart and icons endpoints start with getting the index file. This results in all of those calls happening against a filtered index file. 

### What was fixed, or what changes have occurred 
A `skipFiltering` flag was added so that the filtering of the chart index could be more finely controlled. 

The info endpoint now has `skipFiltering` set to false so that anytime we want to know all of the currently supported charts, charts will get filtered out properly. 

The info, chart and icon endpoints have `skipFiltering` true, because if the chart exist in the repository, we should be able to get the chart info, the chart files and the chart's icon

### Areas or cases that should be tested 
Chart install, edit without upgrading, upgrade, uninstall. This should be done on a chart with and without crds
In rancher v2.6.0 install a chart that is not supported in v2.6.1 (i.e a chart with catalog.cattle.io/rancher-version: "<= 2.6.0-0" in the chart.yaml). After installing, upgrade rancher to v2.6-head, edit your chart installation without upgrading, upgrade and uninstall. 

### What areas could experience regressions ?
All catalog v2 api endpoints had the flag added, so all of the catalog section could. Since we are skipping filtering, it could regress where charts should've been filtered based on rancher-version or kubernetes version but no longer are. 

### Are the repro steps accurate/minimal ?

Yes
https://github.com/rancher/dashboard/issues/4276#issue-1005961299

**Comment by cbron** [comment]
Assigning @jiaqiluo for dev-validate here.

**Comment by jiaqiluo** [comment]
The following validations are done on both the `local` and `downstream` clusters:
- run rancher 2.6.0 and add a downstream cluster 
- add the following GH repo as a repository, it is forked from the release-v2.6 branch  
```
URL:    https://github.com/jakefhyde/charts	
branch: add-min-rancher-version
```
- go to `Charts` -> select only the newly-added repo 
- **Confirmation 1**: for all available charts, all versions show up in the `Chart Versions` list, there is no difference with the released branch
- install the app Istio `1.9.600` into the cluster  
- upgrade rancher to 2.6.1-rc5
- **Confirmation 2**: for all available charts, only the versions introduced in rancher v2.6.0 show up in the `Chart Versions` list; those versions which are also available in rancher v2.5.x are NOT in the list anymore
- **Confirmation 3**: the installed app can be edited on the `Installed App` page: keep the version unchanged, just edit some value. 
- **Confirmation 4**:the installed app can be upgraded on the `Installed App` page: choose the new version from the list
